In [41]:
def read_file(file_name: str) -> str:
    """функция чтения исходного текста из файла"""   
    text = ""     
    with open(file_name) as file:
          text = file.read()
          
    
    return text

In [42]:
def read_frequency_file(file_name: str) -> dict:
    """функция чтения частотного текста из файла""" 
    frequency_dict = {}
    with open(file_name) as file:
        for item in file:
            tmp_str = item.split("=")
            key = tmp_str[0][0]
            value = round(float(tmp_str[1][1:9]), 6)
            frequency_dict.update({key:value})
    return(frequency_dict)


In [43]:
frequency_dict = read_frequency_file("frequency.txt" )


TypeError: read_frequency_file() got an unexpected keyword argument 'encoding'

In [45]:
with open('text_2.txt',  encoding='utf-8') as file:
    text = file.read()
print(text)      


r2Я Ь9К>>К2ЙЬБt ЬrДЫt12<Д>>tr82tМ<КЫ1ХО>12>t<rЬЧ82ЯК КБrtАД-ХО>12ЯЬ2ЙЬБЬrЬ<02Я Ь>О ДЫ>Оr02ЬЕК>ЯКatrД12r2Е0Б0cК<2Ф..КЙОtrЫЬК2ЯЬ2r К<КЫt2БКЙЬБt ЬrДЫtК2>ЛДОЬАЬ2ОКЙ>ОД2ЫЬ2r<К>ОК2>2tМ<КЫКЫtК<2 Д>ЯЬЧЬЛКЫt12ЬО КМЙЬr2>ЬЬОrКО>Оr0Хct32>t<rЬЧД<2r2ЙЬБЬrЬ<2Я Ь>О ДЫ>ОrК2tМ<КЫ1КО>12ОДЙЛК2t2Я КБ>ОДrЧКЫt12>ЛДО832>t<rЬЧЬr2ОЬ2К>О,2r2ЫДbt32ОК <tЫД32bt. ЬЕЬМЫДaКЫt1?2ЫДЯ t<К 2ЯК КБrtЫ0r2>t<rЬЧ2r2tМ2ЙЬЫ9Д2ЙЬБЬrЬАЬ2Я Ь>О ДЫ>ОrД2r2ЫДaДЧЬ2ЙЬБt 0ХcКК2МЫДaКЫtК2<ЬЛКО2r8АЧ1БКО,2ЫК2ЙДЙ2МДЯЧДЫt ЬrДЫЫЬК2tМЫДaДЧ,ЫЬ2Д2ЯЬБ 0АЬ<02>ЬЬОrКО>ОrКЫЫЬ2<КЫ1ХО>12t2ЕtО82ЙЬБt 0ХctК2ЫДbК2 КМ0Ч,Оt 0ХcКК2МЫДaКЫtК2rБЬЕДrЬЙ2ДБДЯОtrЫЬ<2ДЧАЬ tО<К2r>К2<КЫ1КО>12БtЫД<taК>Йt2ЯЬ>ЧК2tМ<КЫКЫt12ЙДЛБЬАЬ2>t<rЬЧД


In [17]:
def frequency_new_text(text: str) -> dict:
    frequency_new_dict ={}
    len_text = len(text)
    for i in text:
        count = 0.00000
        if i not in frequency_new_dict.keys():
            for j in range (len_text):                
                if i == text[j]:
                    count+=1
            frequency_new_dict.update({i: round(float(count/len_text), 6)})


    return frequency_new_dict        


In [18]:
frequency_new_dict = frequency_new_text(text)

In [19]:
frequency_new_dict = dict(sorted(frequency_new_dict.items(), key=lambda item: item[1], reverse=True))
frequency_dict = dict(sorted(frequency_dict.items(), key=lambda item: item[1], reverse=True))
frequency_dict = tuple(frequency_dict.items())
frequency_new_dict = tuple(frequency_new_dict.items())
for i in range (32):
    print(i, ' ', frequency_new_dict[i], '\t\t', frequency_dict[i])

0   ('2', 0.120061) 		 (' ', 0.128675)
1   ('К', 0.092705) 		 ('О', 0.096456)
2   ('Ь', 0.091185) 		 ('И', 0.075312)
3   ('t', 0.069909) 		 ('Е', 0.072292)
4   ('Ы', 0.06231) 		 ('А', 0.064841)
5   ('r', 0.06079) 		 ('Н', 0.06182)
6   ('Д', 0.06079) 		 ('Т', 0.061619)
7   ('>', 0.054711) 		 ('С', 0.051953)
8   ('О', 0.053191) 		 ('Р', 0.040677)
9   (' ', 0.042553) 		 ('В', 0.039267)
10   ('<', 0.039514) 		 ('М', 0.029803)
11   ('Б', 0.027356) 		 ('Л', 0.0294)
12   ('Й', 0.025836) 		 ('Д', 0.026983)
13   ('Я', 0.024316) 		 ('Я', 0.026379)
14   ('1', 0.024316) 		 ('К', 0.025977)
15   ('Ч', 0.021277) 		 ('П', 0.024768)
16   ('М', 0.018237) 		 ('З', 0.015908)
17   ('0', 0.018237) 		 ('Ы', 0.015707)
18   ('Х', 0.010638) 		 ('Ь', 0.015103)
19   ('А', 0.010638) 		 ('У', 0.01329)
20   ('a', 0.010638) 		 ('Ч', 0.011679)
21   ('Л', 0.009119) 		 ('Ж', 0.010673)
22   ('8', 0.007599) 		 ('Г', 0.009867)
23   ('Е', 0.007599) 		 ('Х', 0.008659)
24   ('c', 0.007599) 		 ('Ф', 0.007249)
25   ('3', 0.0060

r2Я Ь9К>>К2ЙЬБt ЬrДЫt12<Д>>tr82tМ<КЫ1ХО>12>t<rЬЧ82ЯК КБrtАД- ХО>12ЯЬ2ЙЬБЬrЬ<02Я Ь>О ДЫ>Оr02ЬЕК>ЯКatrД12r2Е0Б0cК<2Ф..КЙОtrЫЬК2ЯЬ2r К<КЫt2БКЙЬБt ЬrДЫtК2>ЛДОЬАЬ2ОКЙ>ОД2ЫЬ2r<К>ОК2>2tМ<КЫКЫtК<2 Д>ЯЬЧЬЛКЫt12ЬО КМЙЬr2>ЬЬОrКО>Оr0Хct32>t<rЬЧД<2r2ЙЬБЬrЬ<2Я Ь>О ДЫ>ОrК2tМ<КЫ1КО>12ОДЙЛК2t2Я КБ>ОДrЧКЫt12>ЛДО832>t<rЬЧЬr2ОЬ2К>О,2r2ЫДbt32ОК <tЫД32bt. ЬЕЬМЫДaКЫt1?2ЫДЯ t<К 2ЯК КБrtЫ0r2>t<rЬЧ2r2tМ2ЙЬЫ9Д2ЙЬБЬrЬАЬ2Я Ь>О ДЫ>ОrД2r2ЫДaДЧЬ2ЙЬБt 0ХcКК2МЫДaКЫtК2<ЬЛКО2r8АЧ1БКО,2ЫК2ЙДЙ2МДЯЧДЫt ЬrДЫЫЬК2tМЫДaДЧ,ЫЬ2Д2ЯЬБ 0АЬ<02>ЬЬОrКО>ОrКЫЫЬ2<КЫ1ХО>12t2ЕtО82ЙЬБt 0ХctК2ЫДbК2 КМ0Ч,Оt 0ХcКК2МЫДaКЫtК2rБЬЕДrЬЙ2ДБДЯОtrЫЬ<2ДЧАЬ tО<К2r>К2<КЫ1КО>12БtЫД<taК>Йt2ЯЬ>ЧК2tМ<КЫКЫt12ЙДЛБЬАЬ2>t<r ЬЧД

In [51]:
new_text = text
print('1) \' \' = В')
new_text = new_text.replace(frequency_new_dict[9][0], frequency_dict[6][0])
print('2) 2 = \' \'')
new_text = new_text.replace(frequency_new_dict[0][0], frequency_dict[0][0])
print('3) r = c')
new_text = new_text.replace(frequency_new_dict[5][0], frequency_dict[7][0])
print('3) м = з')
new_text = new_text.replace(frequency_new_dict[16][0], frequency_dict[16][0])
print('3) ь = а')
new_text = new_text.replace(frequency_new_dict[2][0], frequency_dict[4][0])
print('3) о = н')
new_text = new_text.replace(frequency_new_dict[8][0], frequency_dict[5][0])
print('3) t = и')
new_text = new_text.replace(frequency_new_dict[3][0], frequency_dict[2][0])
print('3) 1 = я')
new_text = new_text.replace(frequency_new_dict[14][0], frequency_dict[13][0])
print('3) к = е')
new_text = new_text.replace(frequency_new_dict[1][0], frequency_dict[3][0])


print(new_text)

1) ' ' = В
2) 2 = ' '
3) r = c
3) м = з
3) ь = а
3) о = н
3) t = и
3) 1 = я
3) к = е
С ЯТА9Е>>Е ЙАБИТАСДЫИЯ <Д>>ИС8 ИЗ<ЕЫЯХН>Я >И<САЧ8 ЯЕТЕБСИАД-ХН>Я ЯА ЙАБАСА<0 ЯТА>НТДЫ>НС0 АЕЕ>ЯЕaИСДЯ С Е0Б0cЕ< Ф..ЕЙНИСЫАЕ ЯА СТЕ<ЕЫИ БЕЙАБИТАСДЫИЕ >ЛДНААА НЕЙ>НД ЫА С<Е>НЕ > ИЗ<ЕЫЕЫИЕ< ТД>ЯАЧАЛЕЫИЯ АНТЕЗЙАС >ААНСЕН>НС0ХcИ3 >И<САЧД< С ЙАБАСА< ЯТА>НТДЫ>НСЕ ИЗ<ЕЫЯЕН>Я НДЙЛЕ И ЯТЕБ>НДСЧЕЫИЯ >ЛДН83 >И<САЧАС НА Е>Н, С ЫДbИ3 НЕТ<ИЫД3 bИ.ТАЕАЗЫДaЕЫИЯ? ЫДЯТИ<ЕТ ЯЕТЕБСИЫ0С >И<САЧ С ИЗ ЙАЫ9Д ЙАБАСААА ЯТА>НТДЫ>НСД С ЫДaДЧА ЙАБИТ0ХcЕЕ ЗЫДaЕЫИЕ <АЛЕН С8АЧЯБЕН, ЫЕ ЙДЙ ЗДЯЧДЫИТАСДЫЫАЕ ИЗЫДaДЧ,ЫА Д ЯАБТ0АА<0 >ААНСЕН>НСЕЫЫА <ЕЫЯХН>Я И ЕИН8 ЙАБИТ0ХcИЕ ЫДbЕ ТЕЗ0Ч,НИТ0ХcЕЕ ЗЫДaЕЫИЕ СБАЕДСАЙ ДБДЯНИСЫА< ДЧААТИН<Е С>Е <ЕЫЯЕН>Я БИЫД<ИaЕ>ЙИ ЯА>ЧЕ ИЗ<ЕЫЕЫИЯ ЙДЛБААА >И<САЧД


С ЯТА9К>>К ЙАБИТАСДЫИ1 <Д>>ИС8 ИЗ<КЫ1ХН>1 >И<САЧ8 ЯКТКБСИАД-ТХН>1 ЯА ЙАБАСА<0 ЯТА>НТДЫ>НС0 АЕК>ЯКaИСД1 С Е0Б0cК< Ф..КЙНИСЫАК ЯА СТК<КЫИ БКЙАБИТАСДЫИК >ЛДНААА НКЙ>НД ЫА С<К>НК > ИЗ<КЫКЫИК< ТД>ЯАЧАЛКЫИ1 АНТКЗЙАС >ААНСКН>НС0ХcИ3 >И<САЧД< С ЙАБАСА< ЯТА>НТДЫ>НСК ИЗ<КЫ1КН>1 НДЙЛК И ЯТКБ>НДСЧКЫИ1 >ЛДН83 >И<САЧАС НА К>Н, С ЫДbИ3 НКТ<ИЫД3 bИ.ТАЕАЗЫДaКЫИ1? ЫДЯТИ<КТ ЯКТКБСИЫ0С >И<САЧ С ИЗ ЙАЫ9Д ЙАБАСААА ЯТА>НТДЫ>НСД С ЫДaДЧА ЙАБИТ0ХcКК ЗЫДaКЫИК <АЛКН С8АЧ1БКН, ЫК ЙДЙ ЗДЯЧДЫИТАСДЫЫАК ИЗЫДaДЧ,ЫА Д ЯАБТ0АА<0 >ААНСКН>НСКЫЫА <КЫ1ХН>1 И ЕИН8 ЙАБИТ0ХcИК ЫДbК ТКЗ0Ч,НИТ0ХcКК ЗЫДaКЫИК СБАЕДСАЙ ДБДЯНИСЫА< ДЧААТИН<К С>К <КЫ1КН>1 БИЫД<ИaК>ЙИ ЯА>ЧК ИЗ<КЫКЫИ1 ЙДЛБААА >И<СТАЧД